File: src/api-dotnet/Prs.Bootcamp/Data/Models/PurchaseRequest.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Prs.Bootcamp.Data.Models;

public class PurchaseRequest
{
    [Key]
    public int Id { get; set; }

    [Required]
    [StringLength(80)]
    public string Description { get; set; } = null!;

    [Required]
    [StringLength(80)]
    public string Justification { get; set; } = null!;

    [StringLength(80)]
    public string ReasonForRejection { get; set; } = string.Empty;

    public DateTime DateNeeded { get; set; }

    [Required]
    [StringLength(20)]
    public string DeliveryMode { get; set; } = null!;

    [Required]
    [StringLength(10)]
    public string Status { get; set; } = null!;

    [Column(TypeName = "DECIMAL(11,2)")]
    public decimal Total { get; set; }

    public DateTime SubmittedDate { get; set; }

    public int UserId { get; set; }
    public User? UserNavigation { get; set; }

    public void Copy(PurchaseRequest purchaseRequest)
    {
        Id = purchaseRequest.Id;
        UserId = purchaseRequest.UserId;
        Description = purchaseRequest.Description;
        Justification = purchaseRequest.Justification;
        DateNeeded = purchaseRequest.DateNeeded;
        DeliveryMode = purchaseRequest.DeliveryMode;
        Status = purchaseRequest.Status;
        Total = purchaseRequest.Total;
        SubmittedDate = purchaseRequest.SubmittedDate;
        ReasonForRejection = purchaseRequest.ReasonForRejection;
    }
}

File: src/api-java/src/main/java/app/bootcamp/prs/model/PurchaseRequest.java
package app.bootcamp.prs.model;

import jakarta.persistence.*;
import java.math.BigDecimal;
import java.time.ZonedDateTime;
import java.util.List;

@Entity
@Table(name = "purchase_request")
public class PurchaseRequest {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "description", nullable = false)
    private String description;

    @Column(name = "justification", nullable = false)
    private String justification;

    @Column(name = "reason_for_rejection")
    private String reasonForRejection;

    @Column(name = "date_needed", nullable = false)
    private ZonedDateTime dateNeeded;

    @Column(name = "delivery_mode", nullable = false)
    private String deliveryMode;

    @ManyToOne
    @JoinColumn(name = "status_id", nullable = false)
    private RequestStatus status;

    @Column(name = "total", nullable = false)
    private BigDecimal total;

    @Column(name = "submitted_date", nullable = false)
    private ZonedDateTime submittedDate;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private AppUser user;

    @OneToMany(mappedBy = "purchaseRequest", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<PurchaseRequestLineItem> lineItems;

    // Getters and Setters
    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getJustification() {
        return justification;
    }

    public void setJustification(String justification) {
        this.justification = justification;
    }

    public String getReasonForRejection() {
        return reasonForRejection;
    }

    public void setReasonForRejection(String reasonForRejection) {
        this.reasonForRejection = reasonForRejection;
    }

    public ZonedDateTime getDateNeeded() {
        return dateNeeded;
    }

    public void setDateNeeded(ZonedDateTime dateNeeded) {
        this.dateNeeded = dateNeeded;
    }

    public String getDeliveryMode() {
        return deliveryMode;
    }

    public void setDeliveryMode(String deliveryMode) {
        this.deliveryMode = deliveryMode;
    }

    public RequestStatus getStatus() {
        return status;
    }

    public void setStatus(RequestStatus status) {
        this.status = status;
    }

    public BigDecimal getTotal() {
        return total;
    }

    public void setTotal(BigDecimal total) {
        this.total = total;
    }

    public ZonedDateTime getSubmittedDate() {
        return submittedDate;
    }

    public void setSubmittedDate(ZonedDateTime submittedDate) {
        this.submittedDate = submittedDate;
    }

    public AppUser getUser() {
        return user;
    }

    public void setUser(AppUser user) {
        this.user = user;
    }

    public List<PurchaseRequestLineItem> getLineItems() {
        return lineItems;
    }

    public void setLineItems(List<PurchaseRequestLineItem> lineItems) {
        this.lineItems = lineItems;
    }
}